### ISSUE TITLE
Better support for older OOTB images

### ISSUE THREAD

**Issue by harshad16** [opendatahub-io/notebooks#131]
Description of problem:

In an older DS project, many of my workbenches now have an "unknown" image associated:

If I try to edit the project, I can't see which image it was using, I'm forced to choose from the available images. 

And yet, I can still start and use this workbench. 

Prerequisites (if any, like setup, operators/versions):

Have an older cluster with projects that were created a while back. 

Steps to Reproduce
<steps>
Actual results:
unknown image.
unable to know which one it even was. (was it a tensorflow or a pytorch?)
Expected results:
at least know which image it was.
even if I can't select an N-2 image for a fresh project, I should be able to keep using it. 
Reproducibility (Always/Intermittent/Only Once):

I've seen the same thing in other projects too. 

![image-2023-05-29-12-10-38-190](https://github.com/opendatahub-io/notebooks/assets/14028058/84b2a4c6-f0eb-4894-b7d3-8583c7122575)

Build Details:
Workaround:
Additional info:

![image-2023-05-29-12-12-07-256](https://github.com/opendatahub-io/notebooks/assets/14028058/6fb3b7a4-4eed-4df4-a9dc-059b831c2dc3)

These are the images behind the scenes:

bash-4.4 ~ $ oc get statefulsets -n a-great-project -o jsonpath='{range .items[*]}{.spec.template.spec.containers[*].image}{"\n"}{end}'
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/byon-1681497173054:jupyter-datascience-ubi9-python-3.9-20230411 
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/pytorch:py3.8-cuda-11.4.2-2 
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/byon-1679684285597:latest 
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/tensorflow:py3.8-cuda-11.4.2-2 
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/s2i-minimal-notebook:py3.8-v1 
image-registry.openshift-image-registry.svc:5000/redhat-ods-applications/byon-1675352918609:code-server-ubi9-py39_2023b_latest 
bash-4.4 ~ $ 

JIRA: https://issues.redhat.com/browse/RHODS-8933

**Comment by harshad16** [comment]
Based on the discussion, this requires more discussion in term of the annotations of the imagestream
and additional discussion with the dashboard team.

Moving this issue to the next sprint 1.31

**Comment by atheo89** [comment]
The dashboard team is planning on storing the name of the image that we show in the UI on the Notebook CR as an annotation `opendatahub.io/image-display-name`, doing this to address the unknown situation if the source slips.
This coupled with the idea of “still checking” the live value will allow us to know if what they have selected will be re-selectable. If it’s not found (which causes Unknown today) then we can add (deprecated) onto the end of whatever name we held.

The action item for the workbenches is when we move to drop the current N-1, to keep this resource around and flag it so the dashboard to be able to read and ignore it when rendering. 
This can be accomplished by introducing an annotation flag on the tag assigned to the resource we are deprecating. Examples of such annotation flags could be:

-  `image-deleted:true` or `deprecated-image:true ` 
-  `deprecation-text: This notebook is no longer supported; please upgrade to a newer version`

As it stands, the dashboard team is working on [1370](https://github.com/opendatahub-io/odh-dashboard/issues/1370) issue but it is related to the functionality when we are going to slip an old tag, so it is crucial to keep a flag that the dashboard can render and provide the corresponding notification to the user.

Here is the issue to keep tracking this workload https://github.com/opendatahub-io/notebooks/issues/154

**Comment by harshad16** [comment]
Thanks for effectively working on this. 
As the follow up work is created into a separate issue: https://github.com/opendatahub-io/notebooks/issues/154
Closing this issue as done and rest of the work would be carried out on issue #154
